tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc

```diff
@@ -393,28 +420,61 @@ void TRTEngineOp::ExecuteCalibration(OpKernelContext* ctx,
   ExecuteNativeSegment(ctx, helper);
 }
 
-Status TRTEngineOp::VerifyInputShapes(const std::vector<TensorShape>& shapes) {
-  if (shapes.empty()) {
+Status TRTEngineOp::VerifyInputShapes(
+    const std::vector<TensorShape>& input_concrete_shapes) {
+  if (input_concrete_shapes.empty()) {
     return errors::InvalidArgument("Input shapes are empty, for ", name());
   }
-  if (shapes[0].dims() < 1) {
-    return errors::InvalidArgument("Input shapes contain scalar, for ", name(),
-                                   ": ",
-                                   TensorShapeUtils::ShapeListString(shapes));
+
+  if (input_partial_shapes_.size() == 0) {
+    if (!use_implicit_batch_) {
```

Comment:
Can you add a comment to state why it is ok for use_implicit_batch to have empty input_partial_shapes_ and maybe link this to the error message emit fromTRTEngineOp::TRTEngineOp that says this is to support the creation of engine for the TRTEngine op created using an older version of TF-TRT?